service/src/main/java/greencity/service/ubs/UBSManagementServiceImpl.java

```diff
@@ -905,89 +905,104 @@ private void setValueForOrderStatusIsCancelledOrDoneAsTrue(OrderStatusTranslatio
+        }
+        if (nonNull(confirmed)) {
+            for (Map.Entry<Integer, Integer> entry : confirmed.entrySet()) {
+                if (Boolean.TRUE.equals(!updateOrderRepository.ifRecordExist(orderId,
+                    entry.getKey().longValue()))) {
+                    updateOrderRepository.insertNewRecord(orderId,
+                        entry.getKey().longValue());
+                }
                 updateOrderRepository
-                    .updateConfirm(updateOrderDetailDto.getConfirmedQuantity(), updateOrderDetailDto.getOrderId(),
-                        updateOrderDetailDto.getBagId().longValue());
+                    .updateConfirm(entry.getValue(), orderId,
+                        entry.getKey().longValue());
             }
         }
-
-        Order order = orderRepository.getOrderDetails(request.get(0).getOrderId())
-            .orElseThrow(() -> new UnexistingOrderException(
-                ORDER_WITH_CURRENT_ID_DOES_NOT_EXIST + request.get(0).getOrderId()));
-
-        setOrderDetailDto(dto, order, request.get(0).getOrderId(), language);
-        orderRepository.save(order);
-
-        final User currentUser = userRepository.findUserByUuid(uuid)
-            .orElseThrow(() -> new UserNotFoundException(USER_WITH_CURRENT_ID_DOES_NOT_EXIST));
-        collectEventsAboutSetOrderDetails(request, currentUser, language);
-
-        return modelMapper.map(dto, new TypeToken<List<OrderDetailInfoDto>>() {
-        }.getType());
     }
 
-    private void collectEventsAboutSetOrderDetails(List<UpdateOrderDetailDto> dto, User currentUser, String language) {
-        Order order = orderRepository.findById(dto.get(0).getOrderId()).orElseThrow(
+    private void collectEventsAboutSetOrderDetails(Map<Integer, Integer> confirmed, Map<Integer, Integer> exported,
+        Long orderId, User currentUser, String language) {
+        Order order = orderRepository.findById(orderId).orElseThrow(
             () -> new OrderNotFoundException(ORDER_WITH_CURRENT_ID_DOES_NOT_EXIST));
         Long languageId = languageRepository.findIdByCode(language);
         StringBuilder values = new StringBuilder();
-        for (int i = 0; i < dto.size(); i++) {
-            Integer capacity = bagRepository.findCapacityById(dto.get(i).getBagId());
-            StringBuilder bagTranslation = bagTranslationRepository.findNameByBagId(dto.get(i).getBagId(), languageId);
+        int countOfChanges = 0;
+        if (nonNull(exported)) {
+            collectEventAboutExportedWaste(exported, languageId, order, orderId, countOfChanges, values);
+        }
+        if (nonNull(confirmed)) {
+            collectEventAboutConfirmWaste(confirmed, languageId, order, orderId, countOfChanges, values);
+        }
+        if (nonNull(confirmed) || nonNull(exported)) {
+            eventService.save(values.toString(),
+                currentUser.getRecipientName() + "  " + currentUser.getRecipientSurname(), order);
```

Comment:
instead of building fill name here you can introduce method `String getFullName()` as a part of User.

Reply:
Got it. I will add this on the weekend after I implements all  main logic